src/transformers/trainer_tf.py

```diff
@@ -208,6 +248,22 @@ def _prediction_loop(
 
         return PredictionOutput(predictions=preds, label_ids=label_ids, metrics=metrics)
 
+    def _log(self, logs: Dict[str, float]) -> None:
+        if self.epoch_logging is not None:
+            logs["epoch"] = self.epoch_logging
+        if self.global_step is None:
+            # when logging evaluation metrics without training
+            self.global_step = 0
```

Comment:
Here you are also logging `global_step` and `epoch_logging` during the evaluation, it should not. Then these 5 lines can be removed.

Reply:
Because we use `eval_steps` arg, which may be different than `logging_steps`, we need to make sure we log the epoch as it could be associated to a different step.

My example gives below graph:

* where we have less evaluation steps (that could also be different from the logging steps).
* where we also log final evaluation metrics at the end of training -> you can see `eval_loss` is logged further than `loss`

![image](https://user-images.githubusercontent.com/715491/84574274-fd896500-ad6a-11ea-9a96-06e81f211c9d.png)

The step is barely needed as default x-axis and to make sure we have sequential logs (as with Tensorboard) and we just use the current training step for convenience.

When doing evaluation, we have access to `global_step` and `epoch_logging` as they are being saved during the training directly in the object.

I can set `global_step` in `__init__` so it's never `None` and then I can remove lines 254-256.

I still need to log `epoch` in eval (to make the graph above).
In the current logic, I don't log it at all in the case that I'm using `TFTrainer` only for evaluation (no training) as I thought it would not be useful but if you want I can init it to 0 as well. I would still need to set `logs["epoch"] though.

Reply:
Sorry, what I meant is that `logs["epoch"]` can be set in the `train` method and `global_step` in `__init__`. As if you run only the trainer in eval/predict mode you don't need to have a graph depending of epochs and if you set `global_step` in `__init__` you don't have to do this `if`.